### PR TITLE
Case Management and Database templates

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/18548

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [X] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](link)
- [ ] [CHANGELOG](link)

### CODE REVIEW

File: Templates/Integrations/CaseManagement/CaseManagement.py [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+            demisto.setLastRun(last_run)
+        if command in commands:
+            return_outputs(*commands[command](client, demisto.args()))
+    # Log exceptions
+    except Exception as e:
+        err_msg = f'Error in AuthenticationExample Integration [{e}]'
+        return_error(err_msg, error=e)
+
+
+if __name__ == '__builtin__':
```
Comment (Itay4) [inline Templates/Integrations/CaseManagement/CaseManagement.py]:
```suggestion
if __name__ == '__builtin__':
```
```suggestion
if __name__ == 'builtins':
```

File: Templates/Integrations/CaseManagement/CaseManagement.py [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+            if last_fetch_datetime < new_time:
+                last_fetch_datetime = new_time
+    return incidents, {'timestamp': last_fetch_datetime.strftime(TIME_FORMAT)}
+
+
+def main():
+    params = demisto.params()
+    server = params.get('url')
+    use_ssl = not params.get('insecure', False)
+    proxy = params.get('proxy') == 'true'
```
Comment (Itay4) [inline Templates/Integrations/CaseManagement/CaseManagement.py]: hanlde_proxy()

File: Templates/Integrations/CaseManagement/CaseManagement.py [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+
+        Returns:
+            credentials
+        """
+        suffix = 'ticket'
+        params = dict()
+        if limit:
+            params['limit'] = limit
+        elif self._limit:
+            params['limit'] = limit  # type: ignore
```
Comment (Itay4) [inline Templates/Integrations/CaseManagement/CaseManagement.py]: add error code to `type: ignore` as described here https://github.com/demisto/content/pull/4665

File: Templates/Integrations/CaseManagement/CaseManagement.py [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+
+    Args:
+        client: Client object
+        args: Usually demisto.args()
+
+    Returns:
+        Outputs
+    """
+    # Initialize main vars
+    context = dict()
```
Comment (Itay4) [inline Templates/Integrations/CaseManagement/CaseManagement.py]: why is this needed? lets just init it in the condition where you assign it

File: Templates/Integrations/CaseManagement/CaseManagement.py [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+        }
+        human_readable = tableToMarkdown(
+            title, context_entry, headers=['ID', 'Name', 'Timestamp', 'Description', 'Assignee']
+        )
+        return human_readable, context, raw_response
+    else:
+        return f'{INTEGRATION_NAME} - Could not assign users to ticket ID: {ticket_id}', {}, raw_response
+
+
+def build_raw_users_to_context(users: Union[list, dict]):
```
Comment (Itay4) [inline Templates/Integrations/CaseManagement/CaseManagement.py]: this should be in helper functions

File: Templates/Integrations/CaseManagement/CaseManagement.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+commonfields:
+  id: CaseManagement
+  version: -1
+configuration:
+- defaultvalue: https://example.com
+  display: Server URL (e.g. https://example.com)
+  name: url
+  required: true
+  type: 0
+- display: API Token
```
Comment (Itay4) [inline Templates/Integrations/CaseManagement/CaseManagement.yml]: lets remove this

File: Templates/Integrations/Database/Database.yml [hunk 7]
```diff
@@ -0,0 +1,91 @@
+category: Authentication
```
Comment (Itay4) [inline Templates/Integrations/Database/Database.yml]: update category

File: Templates/Integrations/Database/Database.yml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      defaultValue: '10'
+      description: limit number of returned results, to avoid flooding context.
+      isArray: false
+      name: limit
+      required: false
+      secret: false
+    deprecated: false
+    description: Send query to DB
+    execution: false
+    name: database--query
```
Comment (Itay4) [inline Templates/Integrations/Database/Database.yml]:
```suggestion
    name: database-query
```

File: Templates/Integrations/Database/Database.yml [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+- display: Incident type
+  name: incidentType
+  required: false
+  type: 13
+- defaultvalue: https://example.net
+  display: Server URL (e.g. https://example.net)
+  name: url
+  required: true
+  type: 0
+- display: API Token
```
Comment (Itay4) [inline Templates/Integrations/Database/Database.yml]: remove

File: Templates/Integrations/Database/Database.yml [hunk 10]
```diff
@@ -0,0 +1,91 @@
+category: Authentication
+commonfields:
+  id: Database
+  version: -1
+configuration:
+- display: Fetch incidents
```
Comment (Itay4) [inline Templates/Integrations/Database/Database.yml]: fetch params should be the last params

File: Templates/Integrations/Database/Database.yml [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+      isArray: true
+      name: query
+      required: true
+      secret: false
+    - default: false
+      description: Title of each column. if provided, will add a dict to context
+      isArray: false
+      name: columns
+      predefined:
+      - ''
```
Comment (Itay4) [inline Templates/Integrations/Database/Database.yml]: remove empty argument

File: Templates/Integrations/Database/Database.yml [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+      secret: false
+    deprecated: false
+    description: Send query to DB
+    execution: false
+    name: database--query
+    outputs:
+    - contextPath: Database.Result
+      description: Results from query, in a list
+      type: String
+  dockerimage: demisto/python3:3.7.4.977
```
Comment (Itay4) [inline Templates/Integrations/Database/Database.yml]:
```suggestion
  dockerimage: demisto/python3:3.7.5.3066
```

File: Templates/Integrations/CaseManagement/CaseManagement.yml [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+    name: case-management-create-ticket
+  - deprecated: false
+    description: Closes a ticket
+    execution: false
+    name: case-management-close-ticket
+  - deprecated: false
+    description: Assign ticket to user
+    execution: false
+    name: case-management-assign-ticket
+  dockerimage: demisto/python3:3.7.4.977
```
Comment (Itay4) [inline Templates/Integrations/CaseManagement/CaseManagement.yml]:
```suggestion
  dockerimage: demisto/python3:3.7.5.3066
```

File: Templates/Integrations/CaseManagement/CaseManagement.py (hunk 2 above)
Comment (jochman) [reply Templates/Integrations/CaseManagement/CaseManagement.py]: it's part of the BaseClient.

**Comment by Itay4** [comment]
@richardbluestone please review

**Comment by richardbluestone** [comment]
Updated and reviewed docs.
